tests/e2e/vmlistener_test.go

```diff
@@ -0,0 +1,136 @@
+	s.volumeName = inputparams.GetVolumeNameWithTimeStamp("vmlistener_test")
+	s.containerName = inputparams.GetContainerNameWithTimeStamp("vmlistener_test")
+}
+
+func (s *VMListenerTestParams) TearDownTest(c *C) {
+	// Note: no need to remove container, it is already removed as
+	// vm got killed and `--restart=always` flag is not added
+	out, err := dockercli.DeleteVolume(s.dockerHostIP, s.volumeName)
+	c.Assert(err, IsNil, Commentf(misc.FormatOutput(out)))
+}
+
+var _ = Suite(&VMListenerTestParams{})
+
+// Test vmdkops service restart + kill vm process
+// 1. Create a volume
+// 2. Attach it to a container
+// 3. Restart the vmdkops service
+// 4. Verify vmdkops_admin volume ls > verify volume stays as attached
+// 5. Kill the vm
+// 6. Verification: volume status should be detached
+
+func (s *VMListenerTestParams) TestKillVM(c *C) {
+	log.Printf("START: Test vmdkops service restart + kill vm process")
+
+	out, err := dockercli.CreateVolume(s.dockerHostIP, s.volumeName)
+	c.Assert(err, IsNil, Commentf(misc.FormatOutput(out)))
+
+	out, err = dockercli.AttachVolume(s.dockerHostIP, s.volumeName, s.containerName)
+	c.Assert(err, IsNil, Commentf(misc.FormatOutput(out)))
+
+	status := verification.VerifyAttachedStatus(s.volumeName, s.dockerHostIP, s.esxIP)
+	c.Assert(status, Equals, true, Commentf("Volume %s is not attached", s.volumeName))
+
+	// restart vmdkops service
+	out, err = admincli.RestartVmdkopsService(s.esxIP)
+	c.Assert(err, IsNil, Commentf(misc.FormatOutput(out)))
+
+	// make sure volume stays attached after vmdkopsd restart
+	status = verification.VerifyAttachedStatus(s.volumeName, s.dockerHostIP, s.esxIP)
+	c.Assert(status, Equals, true, Commentf("Volume %s is not attached", s.volumeName))
+
+	// make sure vm was powered on
+	powerState := govc.GetVMPowerState(s.dockerHostName)
+	log.Printf("VM[%s]'s current power state is [%s]", s.dockerHostName, powerState)
+	c.Assert(powerState, Equals, powerOnState, Commentf("VM [%s] should be powered on state", s.dockerHostName))
+
+	// grab worldID/vmProcessID
+	processID := esxcli.GetVMProcessID(s.esxIP, s.dockerHostName)
+	log.Printf("VM's process ID is: %s", processID)
+
+	// kill VM
+	isVMKilled := esxcli.KillVM(s.esxIP, processID)
+	c.Assert(isVMKilled, Equals, true, Commentf("Unable to kill VM %s ...", s.dockerHostName))
+	time.Sleep(10 * time.Second)
+
+	// verifies through govc that vm is poweroff
+	powerState = govc.GetVMPowerState(s.dockerHostName)
+	log.Printf("VM[%s]'s current power state is [%s]", s.dockerHostName, powerState)
+	c.Assert(powerState, Equals, powerOffState, Commentf("VM [%s] should be powered off state", s.dockerHostName))
+
+	// status should be detached
+	volAttachStatus := verification.GetVMAttachedToVolUsingAdminCli(s.volumeName, s.esxIP)
+	c.Assert(volAttachStatus, Equals, properties.DetachedStatus, Commentf("Volume %s is still attached", s.volumeName))
+
+	// power on vm
+	govc.PowerOnVM(s.dockerHostName)
+	time.Sleep(35 * time.Second)
```

Comment:
I think you can use misc.SleepForSec() from vmware/docker-volume-vsphere/tests/utils/misc package